plugins/inputs/modbus/modbus.go

```diff
@@ -217,19 +217,29 @@ func (m *Modbus) InitRegister(fields []fieldContainer, name string) error {
 	sort.Slice(addrs, func(i, j int) bool { return addrs[i] < addrs[j] })
 
 	ii := 0
+	maxQuantity := 1
 	var registersRange []registerRange
+	if name == cDiscreteInputs || name == cCoils {
+		maxQuantity = 2000
+	}
+
+	if name == cInputRegisters || name == cHoldingRegisters {
+		maxQuantity = 125
+	}
 
 	// Get range of consecutive integers
 	// [1, 2, 3, 5, 6, 10, 11, 12, 14]
 	// (1, 3) , (5, 2) , (10, 3), (14 , 1)
 	for range addrs {
+		quantity := 1
 		if ii < len(addrs) {
```

Comment:
Not directly related to the patch, but wouldn't it make sense to rather do
```suggestion
		if ii >= len(addrs) {
			break
		}
```

Reply:
I've tested but that change broke some tests

Reply:
Very strange as from a logic point-of-view this should be equivalent. This is not related to the PR anyway, so consider this as resolved.

Reply:
Let me double check that

Reply:
@srebhan  you are right, your suggestion works 👍